lib/validation.js

```diff
@@ -9,55 +9,87 @@ const responseSchema = Symbol('response-schema')
 const headersSchema = Symbol('headers-schema')
 
 const schemas = require('./schemas.json')
+const createMapCache = require('./mapCache')
+
 const inputSchemaError = fastJsonStringify(schemas.inputSchemaError)
 
-function getValidatorForStatusCodeSchema (statusCodeDefinition) {
-  return fastJsonStringify(statusCodeDefinition)
+const getCachedStringify = createMapCache(createStringifier)
```

Comment:
this creates a global cache. This can leak memory (for certain) and create plenty of problems during unit testing,  as they are not isolated anymore.
A better approach (less functional) is to create a `Map` in `fastify` and pass it through the various methods.

Reply:
Maybe use https://www.npmjs.com/package/lru_map

Reply:
Maybe use https://www.npmjs.com/package/lru_map

Reply:
This is not a cache at all, it is just a global state where schemas and stringify functions are added.

Reply:
I ran into that problem during testing for the validation cache - I'll make a `Map` in `fastify` similar to the `allSchemas` that is currently there and pass it in.

Reply:
I've moved the map up to `fastify` and passed it in - I've also added a test to ensure that cross contamination doesn't occur.